ui/app/actions.js

```diff
@@ -904,29 +903,41 @@ function signTokenTx (tokenAddress, toAddress, amount, txData) {
 
 function updateTransaction (txData) {
   log.info('actions: updateTx: ' + JSON.stringify(txData))
-  return (dispatch) => {
+  return dispatch => {
     log.debug(`actions calling background.updateTx`)
-    background.updateTransaction(txData, (err) => {
-      dispatch(actions.hideLoadingIndication())
-      dispatch(actions.updateTransactionParams(txData.id, txData.txParams))
-      if (err) {
-        dispatch(actions.txError(err))
-        dispatch(actions.goHome())
-        return log.error(err.message)
-      }
-      dispatch(actions.showConfTxPage({ id: txData.id }))
+    dispatch(actions.showLoadingIndication())
+
+    return new Promise((resolve, reject) => {
+      background.updateTransaction(txData, (err) => {
+        dispatch(actions.updateTransactionParams(txData.id, txData.txParams))
+        if (err) {
+          dispatch(actions.txError(err))
+          dispatch(actions.goHome())
+          log.error(err.message)
+          return reject(err)
+        }
+
+        resolve(txData)
+      })
     })
+    .then(() => updateMetamaskStateFromBackground())
```

Comment:
Not related to this PR, but we should probably rename `updateMetamaskStateFromBackground` to something like `getUpdatedMetamaskStateFromBackground`. The current name makes it sound like the method does the updating of react state itself.

Reply:
Leaving a TODO here for myself to do this after I finish reviwing:

- [ ] create github issue to rename updateMetamaskStateFromBackground